packages/gatsby/index.d.ts

```diff
@@ -208,8 +208,8 @@ export interface GatsbyNode {
    * See the guide [Creating and Modifying Pages](https://www.gatsbyjs.org/docs/creating-and-modifying-pages/)
    * for more on this API.
    */
-  onCreatePage?<TNode extends object = {}>(
-    args: CreatePageArgs<TNode>,
+  onCreatePage?(
+    args: {page: Page, actions: Actions},
```

Comment:
Rather than change this, could you change `CreatePageArgs` to change the type for `page`. I'd suggest you make onCreatePage generic again, but rather than the Node type argument, make the type argument the context type, which you would pass to your new Page type.

Reply:
Thanks! My mistake. I wasn't clear on the naming convention used here.